Enable broadcastable target to be namespaced by parent

alexandreruban opened the pull request:
Closes #114 

The class method `.broadcast_target_default` now delegates to the instance method `#broadcast_target_default`. It is still equal to `model_name.plural` by default but can now easily be overridden at the instance level. This is especially useful if you want to “namespace” targets by the parent name and id. 

Consider these two models:

```rb
class Message < ApplicationRecord
  has_many :comments 

  # Default target is "messages"
  broadcasts_to ->(message) { :messages }
end
```

```rb 
class Comment < ApplicationRecord
  belongs_to :message

  # Default target is "message_#{message.id}_comments"
  broadcasts_to ->(comment) { [comment.message, :comments] }

  private

  def broadcast_target_default
    "message_#{message.id}_comments"
  end
end
```

Without these changes, if you display mutiple messages and comments on the same page, you will end up with more than one "comments" target. Instead, you want to namespace them by the dom_id of the parent message.

Please tell me if this is OK or if it requires changes! :)

alexandreruban commented on app/models/concerns/turbo/broadcastable.rb:
```diff
@@ -52,21 +52,21 @@ module ClassMethods
     # All default targets will use the return of this method. Overwrite if you want something else than <tt>model_name.plural</tt>.
     def broadcast_target_default
-      model_name.plural
+      ->(broadcastable) { broacastable.send(:broadcast_target_default) }
```
Does it make sense to make instance method `#broadcast_target_default` a public method?

blaknite commented:
Hey @alexandreruban I'd love to see this functionality exist. I hit a bug in my app just now that needed targets scope to the parent model. I've monkey patched `AppicationRecord` with what you've provided here and it works.

As an FYI there's more calls to `broadcast_target_default` within the concern you've edited that will likely explode with the change as is. If I get some time in the next few days I'll flesh those out.

dhh commented:
Don't like the idea of an instance method for broadcast_target_default, but do like the idea that you can pass a proc as a target. Let's just slim it down to just that.

alexandreruban commented:
@dhh thank you for the feedback!

I reverted my changes to only keep the part that enables to pass a proc as a target. Please tell me if it's OK for you now!

dhh commented:
Looks good, just need to add some test coverage 👍

alexandreruban commented:
I added tests for both `Turbo::Broadcastable.broadcasts_to` and `Turbo::Broadcastable.broadcasts` class methods. I added a database in order to test the override with the proc and the whole behaviour with the `after_(create|update|destroy)_commit` callbacks. This is why I created two ActiveRecord models, one with `broadcasts_to` and one with `broadcasts`.

Please tell me if it works for you and thanks again for your feedback! @dhh
